App/Services/SettingsManager.cs

```diff
@@ -1,65 +1,57 @@
+using Google.Protobuf.WellKnownTypes;
 using System;
 using System.Collections.Generic;
 using System.IO;
 using System.Text.Json;
 using System.Text.Json.Serialization;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Xml.Linq;
 
 namespace Coder.Desktop.App.Services;
 
 /// <summary>
 /// Settings contract exposing properties for app settings.
 /// </summary>
-public interface ISettingsManager
+public interface ISettingsManager<T> where T : ISettings, new()
 {
     /// <summary>
-    /// Returns the value of the StartOnLogin setting. Returns <c>false</c> if the key is not found.
+    /// Reads the settings from the file system.
+    /// Always returns the latest settings, even if they were modified by another instance of the app.
+    /// Returned object is always a fresh instance, so it can be modified without affecting the stored settings.
     /// </summary>
-    bool StartOnLogin { get; set; }
-
+    /// <param name="ct"></param>
+    /// <returns></returns>
+    public Task<T> Read(CancellationToken ct = default);
+    /// <summary>
+    /// Writes the settings to the file system.
+    /// </summary>
+    /// <param name="settings">Object containing the settings.</param>
+    /// <param name="ct"></param>
+    /// <returns></returns>
+    public Task Write(T settings, CancellationToken ct = default);
     /// <summary>
-    /// Returns the value of the ConnectOnLaunch setting. Returns <c>false</c> if the key is not found.
+    /// Returns null if the settings are not cached or not available.
     /// </summary>
-    bool ConnectOnLaunch { get; set; }
+    /// <returns></returns>
+    public T? GetFromCache();
```

Comment:
Rather than have this, `Read` should probably return from cache by default. I know this goes against a point I made in a previous review but if we're going with the Read/Write API then it should just be all in

Reply:
Also, Read should probably not fail if it failed to read/parse the file. IDK what good behavior is, but I think for now just returning a default config is OK